src/xrc/src/forex.rs

```diff
@@ -356,26 +365,25 @@ impl ForexRateStore {
         Err(GetForexRateError::InvalidTimestamp(timestamp))
     }
 
-    /// Puts or updates rates for a given timestamp. If rates already exist for the given timestamp, only rates for which a new rate with higher number of sources are replaced.
-    pub(crate) fn put(&mut self, timestamp: u64, rates: ForexMultiRateMap) {
+    /// Puts or updates rates for a given timestamp. If rates already exist for the given timestamp,
+    /// only rates for which a new rate with higher number of sources are replaced.
+    pub(crate) fn put(&mut self, timestamp: u64, mut rates: ForexMultiRateMap) {
```

Comment:
How about removing the USD rates before calling this function so that no mutable reference is required here?
If that's messy (for example because we need to remember to remove it wherever the function is called), we can also leave it the way you proposed.

Reply:
Good point. I moved the removal to the when the retrieval of the rates map from the collector occurs.